apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SpannerDateTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright 2022 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Xunit;
+
+namespace Google.Cloud.Spanner.V1.Tests
+{
+    public class SpannerDateTests
+    {
+        private static readonly string MaxDate = DateTime.MaxValue.Date.ToString(CultureInfo.InvariantCulture);
+
+        private static readonly string MinDate = DateTime.MinValue.Date.ToString(CultureInfo.InvariantCulture);
+
+        public static IEnumerable<object[]> FromDateTimeData()
+        {
+            // Input: DateTime values.
+            yield return new object[] { new DateTime(2022, 4, 18, 0, 0, 0, DateTimeKind.Utc) };
```

Comment:
I'd suggest adding more local and UTC ones for "start of day" as well, to catch a bug of time zone conversions which would currently only be spotted in time zones with a negative UTC offset.

Reply:
Added few more test dates. PTAL and suggest if more dates are needed.

Reply:
I don't think we need quite that many - it was really just "one at the start of the day and one at the end of the day" that are important